ddtrace/internal/telemetry/writer.py

```diff
@@ -165,19 +156,25 @@ def _stop_service(self, *args, **kwargs):
     def add_event(self, payload, payload_type):
         # type: (Dict, str) -> None
         """
-        Adds a Telemetry Request to the TelemetryWriter request buffer
+        Adds a Telemetry event to the TelemetryWriter event buffer
 
-        :param Dict payload: stores a formatted telemetry request
+        :param Dict payload: stores a formatted telemetry event
         :param str payload_type: The payload_type denotes the type of telmetery request.
             Payload types accepted by telemetry/proxy v1: app-started, app-closing, app-integrations-change
         """
-        with self._lock:
-            if not self._enabled:
-                return
-
-            request = self._create_telemetry_request(payload, payload_type, self._sequence)
-            self._sequence += 1
-            self._events_queue.append(request)
+        if self._enabled:
+            event = {
+                "tracer_time": int(time.time()),
+                "runtime_id": get_runtime_id(),
+                "api_version": "v1",
+                "seq_id": next(self._sequence),
+                "debug": str(self._debug).lower(),
```

Comment:
Why are we adding these casts to lowercase string here? Previously we were just using the value of `self._debug` as is. Is there a reason for this change?

Reply:
self._debug is serialized to `True`/`False` by the json encoder but we set the `DD-Telemetry-Debug-Enabled` header to `true`/`false`. This just makes the values consistent but this isn't a "fix" since the backend can handle both values.